x-pack/plugins/security_solution_serverless/server/endpoint/services/set_package_policy_flag.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract, ElasticsearchClient } from '@kbn/core/server';
+import type { PackagePolicyClient } from '@kbn/fleet-plugin/server';
+import type { ListResult, PackagePolicy } from '@kbn/fleet-plugin/common';
+import {
+  FLEET_ENDPOINT_PACKAGE,
+  PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  SO_SEARCH_LIMIT,
+} from '@kbn/fleet-plugin/common';
+
+// set all endpoint policies serverless flag to true
+// required so that endpoint will write heartbeats
+export async function setEndpointPackagePolicyServerlessFlag(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  packagePolicyService: PackagePolicyClient
+): Promise<void> {
+  const perPage: number = SO_SEARCH_LIMIT;
+  let page: number = 1;
+  let endpointPackagesResult: ListResult<PackagePolicy> | undefined;
+
+  while (page === 1 || endpointPackagesResult?.total === perPage) {
```

Comment:
This piece of code seems odd to me, Isn't the first condition (page) going to happen just once?

Reply:
yes, we need it to run once then again for each time there are more pages. but we need it to run at least once.

Reply:
if the `page` purpose is to make it run at least once... do you think it could get easier to understand if we change it to a boolean? we do not really care about the number behind page, right?

Reply:
page is used in the event we have more than 1 page.